array.c

```diff
@@ -1431,25 +1436,59 @@ rb_ary_shift(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.shift    -> obj or nil
- *     ary.shift(n) -> new_ary
+ *     array.shift -> object or nil
+ *     array.shift(n) -> new_array
+ *
+ *  Removes and returns leading elements of +self+;
+ *  shifts elements down so that the first remaining
+ *  element is at index 0.
+ *
+ *  Argument +n+, if given, must be an
+ *  {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects]
+ *
+ *  ---
+ *
+ *  When no argument is given,
+ *  removes and returns the first element:
+ *    a = [:foo, 'bar', 2]
+ *    a.shift # => :foo
+ *    a # => ['bar', 2]
+ *
+ *  Returns +nil+ if +self+ is empty:
+ *    [].shift # => nil
+ *
+ *  ---
+ *
+ *  When argument +n+ is given, removes the first +n+ elements;
+ *  returns those elements in a new \Array:
+ *    a = [:foo, 'bar', 2]
+ *    a.shift(2) # => [:foo, 'bar']
+ *    a # => [2]
  *
- *  Removes the first element of +self+ and returns it (shifting all
- *  other elements down by one). Returns +nil+ if the array
- *  is empty.
+ *  If +n+ is as large as or larger than <tt>self.length</tt>,
+ *  removes all elements; returns those elements in a new \Array:
+ *    a = [:foo, 'bar', 2]
+ *    a.shift(3) # => [:foo, 'bar', 2]
+ *    a # => []
  *
- *  If a number +n+ is given, returns an array of the first +n+ elements
- *  (or less) just like <code>array.slice!(0, n)</code> does. With +ary+
- *  containing only the remainder elements, not including what was shifted to
- *  +new_ary+. See also Array#unshift for the opposite effect.
+ *  If +n+ is 0, returns a new empty \Array;
```

Comment:
Can this fit on a single line instead of 2?

Reply:
Done.